src/main/java/com/ververica/flink/table/gateway/operation/DropViewOperation.java

```diff
@@ -54,10 +58,12 @@ public ResultSet execute() {
 		// all the remaining views are OK, so do the ExecutionContext rebuilding to avoid breaking the view dependency.
 		Environment newEnv = env.clone();
 		if (newEnv.getTables().remove(viewName) != null) {
+			context.getExecutionContext().getTableEnvironment().dropTemporaryView(viewName);
```

Comment:
can we simply this logic: let only `CatalogManager` hold temporary views which means only the new added line is necessary.

Reply:
This deserves a new PR.